app/js/ContentCacheManager.js

```diff
@@ -219,6 +233,28 @@ async function writePdfStream(dir, hash, buffer) {
   }
 }
 
+function getDeadlineChecker(compileTime) {
+  const maxOverhead = Math.min(
+    // Adding 10s to a 40s compile time is OK.
+    compileTime / 4,
+    // Adding 30s to a 120s compile time is not OK, limit to 10s.
+    Settings.pdfCachingMaxProcessingTime
+  )
+
+  const deadline = Date.now() + maxOverhead
+  let lastStage = { stage: 'start', now: Date.now() }
+  return function (stage) {
+    const now = Date.now()
+    if (now > deadline) {
+      throw new TimedOutError(stage, {
```

Comment:
can we put in a counter here to see how many times the deadline checker has been called and return it in the exception -- this will give us an idea of how many operations have been done, e.g. how many blocks have been read from the pdf file by the xref parser.

Reply:
Done in https://github.com/overleaf/clsi/pull/254/commits/756d4a14a18254e217330c606e2a74416af5ff3d